include/roaring/containers/bitset.h

```diff
@@ -64,7 +64,7 @@ bitset_container_t *bitset_container_clone(const bitset_container_t *src);
 void bitset_container_set_range(bitset_container_t *bitset, uint32_t begin,
                                 uint32_t end);
 
-#if defined(CROARING_ASMBITMANIPOPTIMIZATION) && defined(__AVX2__)
+#if defined(CROARING_ASMBITMANIPOPTIMIZATION) && !defined(ROARING_DISABLE_AVX)
```

Comment:
The ASM_SHIFT_RIGHT routine below expects BMI2, I believe.

So not only will it only run under x64, but it also requires BMI2. It is satisfied if __AVX2__ is defined, but the `!defined(ROARING_DISABLE_AVX)` is insufficient. At the very least, we need to check that  `CROARING_IS_X64` is defined, and then, before using it, we need to do a runtime check (e.g., croaring_avx2()). However, it is dubious whether you can do runtime dispatching at the granularity level of such a tiny function and see benefits. Now, it coudb be fruitfully used within a larger function that repeatedly does bitset_container_set, then you can amortize the runtime dispatching. Otherwise, I would simply not bother.

Reply:
i agree, runtime dispatching maybe get little benefits. But for best performance, how about chaging like this:
- #if defined(CROARING_ASMBITMANIPOPTIMIZATION) && defined(__AVX2__)
+ #if defined(CROARING_ASMBITMANIPOPTIMIZATION) && (defined(__AVX2__) || defined(__AVX512__))